mne/epochs.py

```diff
@@ -744,34 +744,26 @@ def _reject_setup(self, reject, flat):
         bad_msg = ('{kind}["{key}"] == {new} {op} {old} (old value), new '
                    '{kind} values must be at least as stringent as '
                    'previous ones')
-        for key in set(reject.keys()).union(old_reject.keys()):
-            old = old_reject.get(key, np.inf)
-            if key in old_reject and key not in reject:
-                # The newly-passed `reject` parameter contains fewer channel
-                # types than the previous one. For the omitted channel types,
-                # simply re-use the old thresholds
-                new = old
-                reject[key] = new
-            else:
-                new = reject[key]
-
-            if new > old:
-                raise ValueError(bad_msg.format(kind='reject', key=key,
-                                                new=new, old=old, op='>'))
-        for key in set(flat.keys()).union(old_flat.keys()):
-            old = old_flat.get(key, -np.inf)
-            if key in old_flat and key not in flat:
-                # The newly-passed `flat` parameter contains fewer channel
-                # types than the previous one. For the omitted channel types,
-                # simply re-use the old thresholds
-                new = old
-                flat[key] = new
-            else:
-                new = flat[key]
 
-            if new < old:
-                raise ValueError(bad_msg.format(kind='flat', key=key,
-                                                new=new, old=old, op='<'))
+        # copy thresholds for channel types that were used previously, but not
+        # passed this time
+        for key in set(old_reject) - set(reject):
+            reject[key] = old_reject[key]
+        # make sure new thresholds are at least as stringent as the old ones
+        for key in reject:
+            if key in old_reject and reject[key] > old_reject[key]:
```

Comment:
```suggestion
        for key in old_reject:
            if reject[key] > old_reject[key]:
```

Reply:
This breaks, unfortunately, in situations where `old_reject` is empty (i.e., if no previous reject has been done) or it doesn't contain a threshold for a channel type that is now appearing in `reject` (i.e., no previous reject for this specific channel type has been done).

I think, therefore, it's okay to keep as-is, and will proceed to merging, ok?

Reply:
ah, yes. I didn't think about old_reject being empty. Will merge.